src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/versionMatcher.hpp

```diff
@@ -0,0 +1,664 @@
+
+            case VersionObjectType::Unspecified:
+                try
+                {
+                    return std::make_shared<VersionObjectCalVer>(version);
+                }
+                catch (const std::runtime_error& e)
+                {
+                    try
+                    {
+                        return std::make_shared<VersionObjectPEP440>(version);
+                    }
+                    catch (const std::runtime_error& e)
+                    {
+                        try
+                        {
+                            return std::make_shared<VersionObjectMajorMinor>(version);
+                        }
+                        catch (const std::runtime_error& e)
+                        {
+                            try
+                            {
+                                return std::make_shared<VersionObjectSemVer>(version);
+                            }
+                            catch (const std::runtime_error& e)
+                            {
+                                throw std::runtime_error {"Error creating VersionObject. Unrecognized type"};
+                            }
+                        }
+                    }
+                }
+
+            case VersionObjectType::CalVer:
+                try
+                {
+                    return std::make_shared<VersionObjectCalVer>(version);
+                }
+                catch (const std::runtime_error& e)
+                {
+                    throw std::runtime_error {
+                        "Error creating VersionObject. Version string dont match the type specified"};
+                }
+
+            case VersionObjectType::PEP440:
+                try
+                {
+                    return std::make_shared<VersionObjectPEP440>(version);
+                }
+                catch (const std::runtime_error& e)
+                {
+                    throw std::runtime_error {
+                        "Error creating VersionObject. Version string dont match the type specified"};
+                }
+
+            case VersionObjectType::MajorMinor:
+                try
+                {
+                    return std::make_shared<VersionObjectMajorMinor>(version);
+                }
+                catch (const std::runtime_error& e)
+                {
+                    throw std::runtime_error {
+                        "Error creating VersionObject. Version string dont match the type specified"};
+                }
+
+            case VersionObjectType::SemVer:
+                try
+                {
+                    return std::make_shared<VersionObjectSemVer>(version);
+                }
+                catch (const std::runtime_error& e)
+                {
+                    throw std::runtime_error {
+                        "Error creating VersionObject. Version string dont match the type specified"};
+                }
+        }
+    }
```

Comment:
```
try {
    switch (type) {
        default: throw std::runtime_error {"Error creating VersionObject. Invalid type"};
        case VersionObjectType::Unspecified:
        case VersionObjectType::CalVer:
            return std::make_shared<VersionObjectCalVer>(version);
        case VersionObjectType::PEP440:
            return std::make_shared<VersionObjectPEP440>(version);
        case VersionObjectType::MajorMinor:
            return std::make_shared<VersionObjectMajorMinor>(version);
        case VersionObjectType::SemVer:
            return std::make_shared<VersionObjectSemVer>(version);
    }
} catch (const std::runtime_error& e) {
    throw std::runtime_error {"Error creating VersionObject. " + e.what()};
}
```

Reply:
Type "Unspecified" has to have a behavior of priority creation of object as is done in the current implementation (has a fallback mechanism).
I don't see a clearer way to implement this one.

Reply:
You right, maybe like this:
```
    std::vector<VersionObjectType> typesToTry;

    if (type == VersionObjectType::Unspecified) {
        typesToTry = {VersionObjectType::CalVer, VersionObjectType::PEP440, VersionObjectType::MajorMinor, VersionObjectType::SemVer};
    } else {
        typesToTry = {type};
    }

    for (const auto& tryType : typesToTry) {
        try {
            switch (tryType) {
                case VersionObjectType::CalVer:
                    return std::make_shared<VersionObjectCalVer>(version);

                case VersionObjectType::PEP440:
                    return std::make_shared<VersionObjectPEP440>(version);

                case VersionObjectType::MajorMinor:
                    return std::make_shared<VersionObjectMajorMinor>(version);

                case VersionObjectType::SemVer:
                    return std::make_shared<VersionObjectSemVer>(version);
            }
        } catch (const std::runtime_error&) {
            continue;
        }
    }

    throw std::runtime_error {"Error creating VersionObject. Unrecognized type"};
```

Reply:
This class needs to have high performance since it will be called many times. Including a loop degrades performance compared to a switch statement.